src/bucket/BucketList.cpp

```diff
@@ -891,6 +891,22 @@ BucketList::scanForEviction(Application& app, AbstractLedgerTxn& ltx,
         auto initialLevel = evictionIter.bucketListLevel;
         auto initialIsCurr = evictionIter.isCurrBucket;
         auto b = getBucketFromIter(evictionIter);
+
+        // Check to see if we can finish scanning the bucket before it receives
+        // an update
+        auto period = bucketUpdatePeriod(evictionIter.bucketListLevel,
+                                         evictionIter.isCurrBucket);
+
+        // Ledgers remaining until the current Bucket changes
+        auto ledgersRemainingToScanBucket = period - (ledgerSeq % period);
+        auto bytesRemaining = b->getSize() - evictionIter.bucketFileOffset;
```

Comment:
Is the bytes remaining calculation here compared to using total bucket size previously done to get more accuracy in case we hit LE limit first during eviction? Also, any particular reason for the code move?

Reply:
Correct, the previous check assumed you read the maximum amount of bytes possible each ledger, which may not be true given the LE limit. The code moved so that we check this every ledger. Previously, we only did this check once when switching to a new bucket. Now, we check every ledger. This will warn more aggressively so it's harder to miss and also allows us to catch the case where we get stuck due to the LE limit.

Reply:
Correct, the previous check assumed you read the maximum amount of bytes possible each ledger, which may not be true given the LE limit. The code moved so that we check this every ledger. Previously, we only did this check once when switching to a new bucket. Now, we check every ledger. This will warn more aggressively so it's harder to miss and also allows us to catch the case where we get stuck due to the LE limit.